### ISSUE TITLE
Bash glob equivalence classes not parsed

### ISSUE THREAD

**Issue by jansorg** [koalaman/shellcheck#2155]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1073
- My shellcheck version (`shellcheck --version` or "online"):  online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
[[ c == [[=c=]] ]] && echo ok
```

#### Here's what shellcheck currently says:
```
Line 2:
[[ c == [[=c=]] ]] && echo ok
^-- SC1073: Couldn't parse this test expression. Fix to allow more checks.
               ^-- SC1020: You need a space before the ]].
               ^-- SC1072: Missing space before ]. Fix any mentioned problems and try again.
```

#### Here's what I wanted or expected to see:
No warning.
`man bash` says:
> Within [ and ], an equivalence class can be specified using the syntax [=c=], which matches all characters with the same  collation weight (as defined by the current locale) as the character c.

ShellCheck currently doesn't seem to parse this syntax.

**Comment by TinCanTech** [comment]
Testing variations, this looks like a `shellcheck` bug:
```
#!/bin/bash

[[ c == [[=c=] ]] && echo ok
```
shellcheck.net
```
No issue detected
```
